Milestone 1 docs (#63)
bitcoinscalinglabs/bitcoin-ipc: docs/deploy-local.md

```sh
ipc-cli wallet list --wallet-type btc
```

## Step 6: Create a child subnet

The next step is to create a subnet under `/b4` — Bitcoin Regtest. Validator 1 will pay the Bitcoin fee, but it could be any of the validators or someone else entirely.

```sh
ipc-cli --config-path ~/.ipc/validator1/config.toml subnet create \
	# general subnet options
	--parent /b4 --min-validators 4 --bottomup-check-period 300 \
	# btc specific options
	btc --min-validator-stake 100000000 --min-cross-msg-fee 10 \
	# comma-separated list of our validators x-only public keys
	--validator-whitelist 5f0dfed3a527ac740c7d4a594cd3aa1059a936187399fc49e3fc6ea6ae177268,851c1bda327584479e98a7c28ea7adc097d290efd105310bcf714231bb99faf4,b15f99928f2478a10c5739a03f5495d342e77352d624e7cc8ebfbded544f9ac0,b45fd52573e8e6bfe0aff82fb228e887fdd92210fe0952ae65a59080fec7e529
```

You should see the subnet ID printed to the console. Let's save the subnet ID for later use.

The create transaction was sent to the mempool, so let's mine a block to include it in the blockchain, afterwhich it'll be picked up by our monitors.

```sh
bitcoin-cli generatetoaddress 1 "$(bitcoin-cli --rpcwallet=default getnewaddress)"
```

> If everything went well, all monitors should have printed the subnet create message to the console.

Reviewer: ```suggestion
All monitors should have printed the subnet create message to the console.
```

## Step 7: Join the subnet

Before we deploy the infrastructure for the subnet, we will have to bootstrap the subnet and join from our validators, putting some initial collateral into the subnet.

> Replace the `--from` field with validator IPC wallet addresses we created in Step 5. Replace the `--subnet` with the subnet ID we got in the previous step.

Reviewer: ```suggestion
In the `--from` argument use the address of the IPC wallet for each validator, as created in Step 5. Replace the `--subnet` with the subnet ID we got in the previous step.
```

```sh

> ipc-cli --config-path ~/.ipc/validator1/config.toml subnet join --from 0x27B60D9f71D6806cCa7D5A92b391093FE100f8e8 --subnet=/b4/t420fdvyrihvwxp5m4ppz2jlwhzq35jaxi4fyints7dwni22fqjz2ftevhzr24e btc --collateral=200000000 --ip 66.222.44.55:8080 --backup-address "$(bitcoin-cli --rpcwallet=validator1 getnewaddress)"

Reviewer: ```suggestion
ipc-cli --config-path ~/.ipc/validator1/config.toml subnet join --from <IPC ADDRESS OF VALIDATOR 1> --subnet=<SUBNET_ ID> btc --collateral=200000000 --ip 66.222.44.55:8080 --backup-address "$(bitcoin-cli --rpcwallet=validator1 getnewaddress)"
```

> ipc-cli --config-path ~/.ipc/validator2/config.toml subnet join --from 0xd9c4C92CA843a53bff146C79B5D32Ca4b9321414 --subnet=/b4/t420fdvyrihvwxp5m4ppz2jlwhzq35jaxi4fyints7dwni22fqjz2ftevhzr24e btc --collateral=110000000 --ip 66.222.44.55:8081 --backup-address "$(bitcoin-cli --rpcwallet=validator2 getnewaddress)"

Reviewer: ```suggestion
ipc-cli --config-path ~/.ipc/validator2/config.toml subnet join --from <IPC ADDRESS OF VALIDATOR 2> --subnet=<SUBNET ID> btc --collateral=110000000 --ip 66.222.44.55:8081 --backup-address "$(bitcoin-cli --rpcwallet=validator2 getnewaddress)"
```

> ipc-cli --config-path ~/.ipc/validator3/config.toml subnet join --from 0x646Aed5404567ae15648E9b9B0004cbAfb126949 --subnet=/b4/t420fdvyrihvwxp5m4ppz2jlwhzq35jaxi4fyints7dwni22fqjz2ftevhzr24e btc --collateral=150000000 --ip 66.222.44.55:8082 --backup-address "$(bitcoin-cli --rpcwallet=validator3 getnewaddress)"

Reviewer: ```suggestion
ipc-cli --config-path ~/.ipc/validator3/config.toml subnet join --from <IPC ADDRESS OF VALIDATOR 3> --subnet=<SUBNET ID> btc --collateral=150000000 --ip 66.222.44.55:8082 --backup-address "$(bitcoin-cli --rpcwallet=validator3 getnewaddress)"
```

> ipc-cli --config-path ~/.ipc/validator4/config.toml subnet join --from 0xBcE2f194e9628E6ae06fa0D85DD57Cd5579213bf --subnet=/b4/t420fdvyrihvwxp5m4ppz2jlwhzq35jaxi4fyints7dwni22fqjz2ftevhzr24e btc --collateral=180000000 --ip 66.222.44.55:8083 --backup-address "$(bitcoin-cli --rpcwallet=validator4 getnewaddress)"

Reviewer: ```suggestion
ipc-cli --config-path ~/.ipc/validator4/config.toml subnet join --from <IPC ADDRESS OF VALIDATOR 4> --subnet=<SUBNET ID> btc --collateral=180000000 --ip 66.222.44.55:8083 --backup-address "$(bitcoin-cli --rpcwallet=validator4 getnewaddress)"
```

```

Let's include the join transactions in the blockchain by mining a block.

```sh
bitcoin-cli generatetoaddress 1 "$(bitcoin-cli --rpcwallet=default getnewaddress)"
```

We should see the monitors print the join messages to the console.

## Step 8: Deploy the infrastructure

Let's export the secret keys of validators:

```sh

> ipc-cli wallet export --wallet-type btc --address 0x27B60D9f71D6806cCa7D5A92b391093FE100f8e8 --hex > ~/.ipc/validator1/validator.sk

Reviewer: ```suggestion
ipc-cli wallet export --wallet-type btc --address <IPC ADDRESS OF VALIDATOR 1> --hex > ~/.ipc/validator1/validator.sk
```

> ipc-cli wallet export --wallet-type btc --address 0xd9c4C92CA843a53bff146C79B5D32Ca4b9321414 --hex > ~/.ipc/validator2/validator.sk

Reviewer: ```suggestion
ipc-cli wallet export --wallet-type btc --address <IPC ADDRESS OF VALIDATOR 2> --hex > ~/.ipc/validator2/validator.sk
```

> ipc-cli wallet export --wallet-type btc --address 0x646Aed5404567ae15648E9b9B0004cbAfb126949 --hex > ~/.ipc/validator3/validator.sk

Reviewer: ```suggestion
ipc-cli wallet export --wallet-type btc --address <IPC ADDRESS OF VALIDATOR 3> --hex > ~/.ipc/validator3/validator.sk
```

> ipc-cli wallet export --wallet-type btc --address 0xBcE2f194e9628E6ae06fa0D85DD57Cd5579213bf --hex > ~/.ipc/validator4/validator.sk

Reviewer: ```suggestion
ipc-cli wallet export --wallet-type btc --address <IPC ADDRESS OF VALIDATOR 4> --hex > ~/.ipc/validator4/validator.sk
```

```

Let's start our first validator which the rest of the validators will bootstrap from. Make sure you have docker running before running this command.

```sh
cargo make --makefile infra/fendermint/Makefile.toml \
    -e NODE_NAME=validator-1 \

>     -e SUBNET_ID=/b4/t420fdvyrihvwxp5m4ppz2jlwhzq35jaxi4fyints7dwni22fqjz2ftevhzr24e \

Reviewer: ```suggestion
    -e SUBNET_ID=<SUBNET ID> \
```
